src/makeCalculator.js

```diff
@@ -1,43 +1,68 @@
 'use strict';
 
-/**
- * Another calculator. Now the task is more difficult.
- * Create a `makeCalculator` function that returns an object that
- * has the following fields:
- *  - Methods: `add`, `subtract`, `multiply`, `divide`, `reset`, `operate`.
- *  - The `result` property is initially 0.
- *
- * How the calculator will work:
- * - Each `operate` call takes a callback and a number and sets the
- *   appropriate value to the `result` property.
- * - The `reset` method resets `result` value to 0.
- * - `add`, `subtract`, `multiply`, `divide` are passed as callbacks to
- *   `operate` method
- * - The `operate` and `reset` methods can be called in a chain.
- *
- * Example:
- * const calculator = makeCalculator();
- *
- * calculator.operate(calculator.add, 21)
- * calculator.result === 21
-
- * calculator.reset()
- * calculator.result === 0
-
- * calculator
- *  .operate(calculator.add, 10)
- *  .reset()
- *  .operate(calculator.subtract, 20)
- *  .operate(calculator.divide, 5)
- *  .operate(calculator.multiply, 7)
- *
- * calculator.result === -28
- *
- *
+/*
  * @return {object}
  */
+
 function makeCalculator() {
-  // write code here
+  const operates = {
+
+    // змінна в якій буде зберігатись результат
+    resultMathOperations: 0,
+
+    // геттер який повертає результат , тобто нашу змінну з результатом
+    get result() {
+      return this.resultMathOperations;
+    },
+
+    // сеттер - який буде змінювати нашу результуючу змінну
+    set changeResult(value) {
+      this.resultMathOperations = value;
+    },
+
+    // дана функція приймає (operation) і число (value) - для маніпуляцій з ним
+    operate(operation, value) {
+      this.changeResult = operation(this.resultMathOperations, value);
+
+      return this;
+    },
+
+    // Функціі ... , які прииймають початкове(або попереднє) значення
+    // (а саме результат) та число яке передається (value),
+    // та вертає результат 'додавання та інші'
+    add(prevValue, value) {
+      const result = prevValue + value;
+
+      return result;
+    },
+
+    subtract(prevValue, value) {
+      const result = prevValue - value;
+
+      return result;
+    },
+
+    multiply(prevValue, value) {
+      const result = prevValue * value;
+
+      return result;
+    },
+
+    divide(prevValue, value) {
+      const result = prevValue / value;
+
+      return result;
+    },
```

Comment:
It is necessary to provide for a check by 0, because according to the rules of mathematics, it is not possible to divide by 0